README.md

```diff
@@ -162,6 +168,139 @@ Click on the button below to deploy this demo to the IBM Cloud.
+
+### Creating a project
+
+After logging in with `oc login`, ensure that you have a project set up. If not, create one as follows:
+
+```
+oc new-project nlu-project --display-name="NLU Code Pattern"
+```
+
+Ensure that your current project is set.
+
+```bash
+oc project nlu-project
+```
+
+### Automatic deployment
+
+Run the deploy.sh script:
+
+```
+sh openshift/deploy.sh
+```
+
+The script will run until the app is deployed and available.
+
+### Manual deployment
+
+<details>
+<summary>Manually deploy to openshift</summary>
+
+#### Prepare the required parameters
+
+The template for this example is located at [template.yaml](openshift/template.yaml).
+
+First, list the available parameters:
+
+```bash
+oc process --parameters -f openshift/template.yaml
+```
+
+The output will look like:
+
+```
+NAME                                          DESCRIPTION                                                                                               GENERATOR           VALUE
+APP_NAME                                      The name assigned to all of the frontend objects defined in this template.                                                    nlu-code-pattern
+NATURAL_LANGUAGE_UNDERSTANDING_AUTH_TYPE      How to authenticate with the NLU service, possible values are: iam, cp4d, bearerToken                                         bearerToken
+NATURAL_LANGUAGE_UNDERSTANDING_BEARER_URL     The service URL.
+NATURAL_LANGUAGE_UNDERSTANDING_BEARER_TOKEN   Cloud Pak for Data bearer token.
+NATURAL_LANGUAGE_UNDERSTANDING_APIKEY         IBM Cloud IAM apikey from the service instance page.
+NATURAL_LANGUAGE_UNDERSTANDING_USERNAME       Cloud Pak for Data username of user with access to the service instance.
+NATURAL_LANGUAGE_UNDERSTANDING_PASSWORD       Cloud Pak for Data password of user with access to the service instance.
+NAMESPACE                                     The OpenShift Namespace where the ImageStream resides.                                                                        openshift
+APPLICATION_DOMAIN                            The exposed hostname that will route to the Node.js service, if left blank a value will be defaulted.                         nlu-code-pattern
+SOURCE_REPOSITORY_URL                         The URL of the repository with your application source code.                                                                  https://github.com/watson-developer-cloud/natural-language-understanding-code-pattern
+SOURCE_REPOSITORY_REF                         Set this to a branch name, tag or other ref of your repository if you are not using the default branch.
+```
+
+Depending on where you service instance is running you may have to provide different parameters to this application. Follow the [instructions](#manual-deployment) on how to run the application locally and create the `.env` file with the service credentials. The same values will be used when deploying to OpenShift.
```

Comment:
There's a small typo in this section. It should be:

"Depending on where **your** service instance..."